docs/guides/serverless-shared-migration-guide.md

```diff
@@ -0,0 +1,266 @@
+
+    Verify that your data is present within the Flex cluster before proceeding.
+4. Delete the Serverless Instance by running a destroy command against it:
+
+    `terraform destroy -target=mongodbatlas_serverless_instance.this`
+
+ 5. You may now safely remove the resource block for the Serverless Instance from your `.tf` file.
+
+### Post-Autoconversion Migration Procedure
+
+Given your Serverless Instance fits the constraints of a Flex cluster, it will automatically convert in March 2025 into a Flex cluster in Atlas, retaining all data.
+
+The following steps resolve the configuration drift in Terraform and does not affect the underlying cluster infrastructure:
+
+1. Find the import IDs of the Flex clusters: `{PROJECT_ID}-{CLUSTER_NAME}`, such as `664619d870c247237f4b86a6-flexClusterName`
+2. Add an import block per cluster to one of your `.tf` files:
+    ```terraform
+    import {
+    to = mongodbatlas_flex_cluster.this
+    id = "664619d870c247237f4b86a6-flexClusterName" # from step 1
+    }
+    ```
+3. Run `terraform plan -generate-config-out=flex_cluster.tf`. This should generate a `flex_cluster.tf` file with your Flex cluster in it.
+4. Run `terraform apply`. You should see the resource(s) imported: `Apply complete! Resources: 1 imported, 0 added, 0 changed, 0 destroyed.`
+5. Remove the "default" fields. Many fields of this resource are optional. Look for fields with a `null` or `0` value.
+6. Re-use existing [Terraform expressions](https://developer.hashicorp.com/terraform/language/expressions). All fields in the generated configuration have static values. Look in your previous configuration for:
+   - variables, for example: `var.project_id`
+   - Terraform keywords, for example: `for_each`, `count`, and `depends_on`
+7. Update the references from your previous cluster resource: `mongodbatlas_serverless_instance.this.X` to the new `mongodbatlas_flex_cluster.this.X`.
+8. Update any shared-tier data source blocks to refer to `mongodbatlas_flex_cluster`.
+9. Replace your existing clusters with the ones from `flex_cluster.tf` and run `terraform state rm mongodbatlas_serverless_instance.this`. Without this step, Terraform creates a plan to delete your existing cluster.
+10.  Remove the import block created in step 2.
+11.  Re-run `terraform plan` to ensure you have no planned changes: `No changes. Your infrastructure matches the configuration.` 
+
+
+## From Serverless to Dedicated 
+**Please note your Serverless instance will need to migrate to Decidated if it meets the following requirements:**
+- \>= 5GB of data
+- needs privatelink or continuous backup
+- \> 500 ops/sec consistently.
+
+You cannot migrate from Serverless to Dedicated using the Terraform provider.
+
+### Pre-Autoconversion Migration Procedure
```

Comment:
same as before: post- before pre- since post- is recommended

Reply:
Done